litgpt/utils.py

```diff
@@ -516,3 +516,10 @@ def instantiate_torch_optimizer(optimizer, model_parameters, **kwargs):
         optimizer["init_args"].update(kwargs)
         optimizer = instantiate_class(model_parameters, optimizer)
     return optimizer
+
+
+def extend_checkpoint_dir(checkpoint_dir: Path) -> Path:
+    new_checkpoint_dir = Path(checkpoint_dir)  # duplicate Path object to not overwrite the original
+    if not new_checkpoint_dir.is_dir() and new_checkpoint_dir.parts[0] != "checkpoints" and not new_checkpoint_dir.is_absolute():
+        new_checkpoint_dir = "checkpoints" / new_checkpoint_dir
```

Comment:
Shouldn't this also check that ./checkpoints exists? If it doesn't exist, there is no point in adding the prefix, right? More generally, IMO, should we only return the extended path if it exists? This would help with error messages down the line pointing to paths the user didn't pass in.

Reply:
How about something like this? (may have typos)
```py
def extend_checkpoint_dir(checkpoint_dir: Path) -> Path:
    new_checkpoint_dir = "checkpoints" / checkpoint_dir
	if not checkpoint_dir.is_dir() and checkpoint_dir.parts[0] != "checkpoints" and not checkpoint_dir.is_absolute() and new_checkpoint_dir.exists():
		return  new_checkpoint_dir
	return checkpoint_dir
```

Reply:
Yes, this is nice and should probably catch all the weird cases! Thanks!

Reply:
I reformatted this a bit so it's more readable without scrolling and added the tests. Thanks again!